src/main/java/org/mariadb/jdbc/MariaDbStatement.java

```diff
@@ -49,33 +49,27 @@
 
 package org.mariadb.jdbc;
 
-import org.mariadb.jdbc.internal.packet.dao.parameters.ParameterHolder;
-import org.mariadb.jdbc.internal.queryresults.*;
 import org.mariadb.jdbc.internal.protocol.Protocol;
 import org.mariadb.jdbc.internal.queryresults.ExecutionResult;
 import org.mariadb.jdbc.internal.queryresults.MultiIntExecutionResult;
 import org.mariadb.jdbc.internal.queryresults.SingleExecutionResult;
 import org.mariadb.jdbc.internal.queryresults.resultset.MariaSelectResultSet;
 import org.mariadb.jdbc.internal.util.ExceptionMapper;
 import org.mariadb.jdbc.internal.util.dao.QueryException;
-import org.mariadb.jdbc.internal.util.scheduler.DynamicSizedSchedulerInterface;
 import org.mariadb.jdbc.internal.util.scheduler.SchedulerServiceProviderHolder;
 
-import org.mariadb.jdbc.internal.protocol.Protocol;
-
 import java.io.IOException;
 import java.io.InputStream;
 import java.sql.*;
 import java.util.*;
-import java.util.concurrent.ExecutionException;
-import java.util.concurrent.Future;
-import java.util.concurrent.TimeUnit;
+import java.util.concurrent.*;
 import java.util.concurrent.locks.ReentrantLock;
 
 
 public class MariaDbStatement implements Statement, Cloneable {
-    private static final DynamicSizedSchedulerInterface timeoutScheduler =
-            SchedulerServiceProviderHolder.getScheduler(0);
+    //timeout scheduler
+    private static volatile ScheduledThreadPoolExecutor timeoutScheduler = null;
```

Comment:
Why lazily construct this?  It should not start any threads until it's used anyways.  Seems like it's worth the simpler runtime logic to not lazily construct it.

Also is there any reason not to set this to the interface `ScheduledExecutorService` so that other implementations can be used?